assignments/this.js

```diff
@@ -1,26 +1,61 @@
 /* The for principles of "this";
 * in your own words. explain the four principle for the "this" keyword below.
 *
-* 1. 
-* 2. 
-* 3. 
-* 4. 
+* 1. Window binding means that the 'this' keyword binds to the window object (also
+    known as default binding)
+* 2. Implicit binding means that the 'this' keyword binds implicitly (automatically)
+to its variable object.
+* 3. Explicit binding is when the 'this' keyword is explicitly assigned to an object
+of the programmer's choice
+* 4. New binding happens when a new object is created, and the 'this' keyword gets
+assigned to that empty object before it is initialized.
 *
 * write out a code example of each explanation above
 */
 
 // Principle 1
 
 // code example for Window Binding
+console.log(this);
 
 // Principle 2
 
 // code example for Implicit Binding
+function test (){
+    console.log(this.name);
+}
 
+const objTest = {
+    name: "Mo Money",
+    test: test
+};
+
+objTest.test();
 // Principle 3
 
 // code example for New Binding
+function testFunc (){
```

Comment:
by convention, Constructor functions are named with CapitalCase. this does not affect the way the code functions, but it provides a helpful visual cue to keep your code standardized and thereby easier for you to read.

```function TestFunc() {```